src/integrated/land.jl

```diff
@@ -36,6 +36,58 @@ struct LandModel{
     canopy::VM
     "The snow model to be used"
     snow::SnM
+    function LandModel{FT}(
+        canopy::VM,
+        snow::SnM,
+        soil::SM,
+        soilco2::MM,
+    ) where {
+        FT,
+        MM <: Soil.Biogeochemistry.SoilCO2Model{FT},
+        SM <: Soil.EnergyHydrology{FT},
+        VM <: Canopy.CanopyModel{FT},
+        SnM <: Snow.SnowModel{FT},
+    }
+        prognostic_land_components = (:canopy, :snow, :soil, :soilco2)
```

Comment:
I double checked, and this order looks OK. (but it does not match the order of ClimaLand.land_components()` which is intentional but not desirable!)

Reply:
Hmm yeah - I guess we should update `land_components()` to use alphabetical order too

Reply:
[We cant](https://github.com/CliMA/ClimaLand.jl/issues/919) but we can discuss this and reprioritize if needed!